chirp/drivers/kg935g.py

```diff
@@ -597,32 +893,42 @@ def _upload(self):
         """Talk to a wouxun KG-935G and do a upload"""
         try:
             self._identify()
-            self._do_upload(0, 32768, 64)
+            self._do_upload()
         except errors.RadioError:
             raise
         except Exception as e:
             raise errors.RadioError("Failed to communicate with radio: %s" % e)
         return
 
-    def _do_upload(self, start, end, blocksize):
-        ptr = start
-        for i in range(start, end, blocksize):
-            req = struct.pack('>H', i)
-            chunk = self.get_mmap()[ptr:ptr + blocksize]
-            self._write_record(CMD_WR, req + chunk)
-            LOG.debug(util.hexprint(req + chunk))
-            cserr, ack = self._read_record()
-            LOG.debug(util.hexprint(ack))
-            j = struct.unpack('>H', ack)[0]
-            if cserr or j != ptr:
-                raise Exception("Radio did not ack block %i" % ptr)
-            ptr += blocksize
-            if self.status_fn:
-                status = chirp_common.Status()
-                status.cur = i
-                status.max = end
-                status.msg = "Cloning to radio"
-                self.status_fn(status)
+    def _do_upload(self):
+        cfgmap = config_map_935G
```

Comment:
I'm missing something. It seems like you're changing `do_upload()` so it can handle multiple regions in the map to upload, but you've only got one map for the 935G. Did the 935G not really care but the 935G plus *does*, yet the same map works for both?

Reply:
The 935G driver originally just fully uploaded all 0x8000 bytes.   This change makes the driver only update the values in known memory areas.  The 2 radios have slightly different maps but the memory regions that contain the known values is the same.  Since both radios identify the same way, I don't know what is in the unknown areas and think it is good to only upload what is known.

Reply:
If you mean why am I setting cfgmap to the 935G map....when I first got the plus, I was prepared to need 2 maps and implemented an IF --Else to assign cfg map the right map.   Since it isn't needed, I just left the local assignment and removed the IF....

Reply:
Yeah, this is all what I meant, just checking.